### PR TITLE
editoast: add errors and warnings for catenary part 1

### LINKED ISSUE #4121: editoast: Add warnings and errors in electrification zone edition
## Description

Editoast should trigger a list of warnings and errors for catenaries

## AC

- This is a list of warnings:
  - No track range (Object)
  - Overlapping catenary (Global)
- This is a list of errors:
  - Invalid reference (`track_ranges` field) (object)
  - Invalid range (`track_ranges` field) (object)

### Implementation

- Create a new `InfraErrorType`, `OverlappingCatenary`.
  - Add it to `schema/errors`, `openapi.yml` and python infra_schema.
- Create an SQL script `generated_data/errors/sql/catenaries_insert_errors.sql`
- Add to the `generated_data/errors/mod.rs:generate` the new error generation.
- Add the `generated_data/errors/catenaries.rs`
  - No need to use context
  - We must test each error generator with unit test

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #4121

### CODE REVIEW

File: editoast/src/generated_data/error/mod.rs
Comment (flomonster) [inline editoast/src/generated_data/error/mod.rs:1]: You should also add a check to the `small_infra_cache_validation` test.

Comment (younesschrifi) [reply editoast/src/generated_data/error/mod.rs:1]: Done !

File: editoast/src/generated_data/error/sql/catenaries_insert_errors.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        schematic,
+        information
+    )
+SELECT $1 AS infra_id,
+    catenaries.geographic,
+    catenaries.schematic,
+    errors.information
+FROM errors
+    LEFT JOIN osrd_infra_catenarylayer AS catenaries ON catenaries.obj_id = information->>'obj_id'
+    AND catenaires.infra_id = $1
```
Comment (flomonster) [inline editoast/src/generated_data/error/sql/catenaries_insert_errors.sql]:
```suggestion
    AND catenaries.infra_id = $1
```

**Review by flomonster — CHANGES_REQUESTED** [review]

**Review by flomonster — APPROVED** [review]
Tested. LGTM

**Pull request merged**